addon/hint/show-hint.js

```diff
@@ -229,30 +229,41 @@
       elt.hintId = i;
     }
 
+    var container = completion.options.container || ownerDocument.body;
     var pos = cm.cursorCoords(completion.options.alignWithWord ? data.from : null);
     var left = pos.left, top = pos.bottom, below = true;
-    hints.style.left = left + "px";
-    hints.style.top = top + "px";
+    var offsetLeft = 0, offsetTop = 0;
+    if(container !== ownerDocument.body) {
+      // We offset the cursor position because left and top are relative to the offsetParent's top left corner.
+      var offsetParent = ['absolute', 'relative', 'fixed'].includes(parentWindow.getComputedStyle(container).position) ? container : container.offsetParent;
```

Comment:
Another nitpick: `includes` doesn't exist on all browsers that we support. Use `indexOf` instead, or a regexp.

Also, a space after `if` would make the code look more like the rest of the codebase.